lib/beacon/multicast.go

```diff
@@ -8,39 +8,198 @@ package beacon
 
 import (
 	"errors"
+	"fmt"
 	"net"
+	"time"
 
+	"github.com/thejerf/suture"
 	"golang.org/x/net/ipv6"
 )
 
 type Multicast struct {
-	conn   *ipv6.PacketConn
+	*suture.Supervisor
```

Comment:
Maybe. I'll double check that it quits and restarts properly on timeout errors etc but I think it's supposed to, yes.